include/revng/ABIAnalyses/Analyses.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/ABIAnalyses/Generated/DeadRegisterArgumentsOfFunction.h"
+#include "revng/ABIAnalyses/Generated/DeadReturnValuesOfFunctionCall.h"
+#include "revng/ABIAnalyses/Generated/RegisterArgumentsOfFunctionCall.h"
+#include "revng/ABIAnalyses/Generated/UsedArgumentsOfFunction.h"
+#include "revng/ABIAnalyses/Generated/UsedReturnValuesOfFunction.h"
+#include "revng/ABIAnalyses/Generated/UsedReturnValuesOfFunctionCall.h"
+
+namespace ABIAnalyses {
+namespace DeadRegisterArgumentsOfFunction {
+using namespace llvm;
+
+DenseMap<const GlobalVariable *, State>
```

Comment:
What's `State`?
Consider that `DenseMap` are designed to be used to map register-side stuff to register-stuff (typically, pointers to pointers). If `State` is larger than that, go for a `std::map`.

Reply:
state is the `model::RegisterState::Values` enum